pkg/observability/observability.go

```diff
@@ -7,16 +7,19 @@ import (
 
 	"go.uber.org/zap"
 	"knative.dev/eventing/pkg/tracing"
+	"knative.dev/pkg/configmap"
 	"knative.dev/pkg/injection/sharedmain"
 	"knative.dev/pkg/logging"
 	"knative.dev/pkg/metrics"
+	"knative.dev/pkg/profiling"
 	tracingconfig "knative.dev/pkg/tracing/config"
 )
 
-// SetupDynamicConfig sets up logging, metrics, and tracing by watching observability
+// SetupDynamicConfigOrDie sets up logging, metrics, and tracing by watching observability
 // configmaps. Returns an updated context with logging and function to flush telemetry which should
 // be called before exit.
-func SetupDynamicConfig(ctx context.Context, componentName string) (context.Context, func(), error) {
+// The input context should have KubeClient injected.
+func SetupDynamicConfigOrDie(ctx context.Context, componentName string) (context.Context, *configmap.InformedWatcher, *profiling.Handler, func()) {
```

Comment:
nit: I feel a more idiomatic name could be `MustSetupDynamicConfig`.

Reply:
I think I will keep `SetupDynamicConfigOrDie`, because this is consistent with other helper functions we call in sharedmain.